gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp

```diff
@@ -258,6 +263,57 @@ TEST_F(GazeboRosWheelSlipTest, TestSetParameters)
   }
 }
 
+class GazeboRosWheelSlipPublisherTest : public gazebo::ServerFixture
+{
+};
+
+TEST_F(GazeboRosWheelSlipPublisherTest, Publishing)
+{
+  // Load test world and start paused
+  this->Load("worlds/gazebo_ros_wheel_slip.world", true);
+
+  // World
+  auto world = gazebo::physics::get_world();
+  ASSERT_NE(nullptr, world);
+
+  // Create node and executor
+  auto node = std::make_shared<rclcpp::Node>("gazebo_ros_joint_state_publisher_test");
+  ASSERT_NE(nullptr, node);
+
+  rclcpp::executors::SingleThreadedExecutor executor;
+  executor.add_node(node);
+
+  // Create subscriber
+  gazebo_msgs::msg::WheelSlip::SharedPtr latestMsg;
+  auto sub = node->create_subscription<gazebo_msgs::msg::WheelSlip>(
+    "trisphere_cycle_slip/wheel_slip", rclcpp::QoS(1),
+    [&latestMsg](const gazebo_msgs::msg::WheelSlip::SharedPtr msg) {
+      latestMsg = msg;
+    });
+
+  // Spin until we get a message or timeout
+  auto startTime = std::chrono::steady_clock::now();
+  while (latestMsg == nullptr &&
+    (std::chrono::steady_clock::now() - startTime) < std::chrono::seconds(2))
+  {
+    world->Step(100);
+    executor.spin_once(100ms);
+    gazebo::common::Time::MSleep(100);
+  }
+
+  // Check that we receive the latest joint state
+  ASSERT_NE(nullptr, latestMsg);
+
+  ASSERT_LT(0u, latestMsg->name.size());
+  ASSERT_GT(4u, latestMsg->name.size());
+  ASSERT_EQ(latestMsg->name.size(), latestMsg->lateral_slip.size());
+  ASSERT_EQ(latestMsg->name.size(), latestMsg->longitudinal_slip.size());
+
+  EXPECT_NEAR(0.0, latestMsg->lateral_slip[0], tol);
+  EXPECT_NE(0.0, latestMsg->longitudinal_slip[0]);
+  EXPECT_GT(0.7, fabs(latestMsg->longitudinal_slip[0]));
```

Comment:
with those changes to the joint name, I'm seeing all three longitudinal wheel slips near `0.2`. How do you feel about replacing this `EXPECT_GT` with an `EXPECT_NEAR(0.2, ...`? You could repeat it for all 3 slip values as well

Reply:
Fixed in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1331/commits/6530b1fa1cbe8db13647e032167c2e00ef080758.